hive-metastore/src/test/java/org/apache/iceberg/hive/HiveTableTest.java

```diff
@@ -298,14 +308,43 @@ public void testFailure() throws TException {
   }
 
   @Test
-  public void testListTables() {
+  public void testListTables() throws TException, IOException {
     List<TableIdentifier> tableIdents = catalog.listTables(TABLE_IDENTIFIER.namespace());
     List<TableIdentifier> expectedIdents = tableIdents.stream()
         .filter(t -> t.namespace().level(0).equals(DB_NAME) && t.name().equals(TABLE_NAME))
         .collect(Collectors.toList());
 
     Assert.assertEquals(1, expectedIdents.size());
     Assert.assertTrue(catalog.tableExists(TABLE_IDENTIFIER));
+
+    // create a hive table
+    String hiveTableName = "test_hive_table";
+    org.apache.hadoop.hive.metastore.api.Table hiveTable = createHiveTable(hiveTableName);
+    metastoreClient.createTable(hiveTable);
+
+    List<TableIdentifier> tableIdents1 = catalog.listTables(TABLE_IDENTIFIER.namespace());
+    Assert.assertEquals("should only 1 iceberg table .", 1, tableIdents1.size());
+    Assert.assertTrue(catalog.tableExists(TABLE_IDENTIFIER));
+    metastoreClient.dropTable(DB_NAME, hiveTableName);
+  }
+
+  private org.apache.hadoop.hive.metastore.api.Table createHiveTable(String hiveTableName) throws IOException {
```

Comment:
a simpler way IMO could have been to use HiveCatalog and Iceberg apis  to create a table and then drop the iceberg metadata from Hive table properties. Though I guess this is fine too.

Reply:
I think we can remove the properties from iceberg table, but it is not a correct hive table. Although the unit test can pass, I am not sure whether it makes sense.